### ISSUE TITLE
[P1] Original code returns inconsistent validation results

### ISSUE THREAD

**Issue by nitaylevy** [stanfordnlp/pyreft#176]
Hi, I'm running the following code from your original code, and for some reason, the in-training validation scores are better than the final test result. Is this a bug? If not, why are they different, because the split is 50/50?
Also, is it safe to assume that the paper results are based on the final test after training?

```
!python task_steer.py \
    -task glue -train_dataset mrpc -model FacebookAI/roberta-base \
    -seed 42 \
    -l all -r 1 -p f3 -e 40 -lr 3e-4 \
    -type ConditionedSourceLowRankRotatedSpaceIntervention \
    -gradient_accumulation_steps 1 \
    -batch_size 32 -eval_batch_size 32 \
    -test_split validation -max_length 256 \
    --metric_for_best_model accuracy \
    --dropout 0.05 --weight_decay 0.00000 \
    --warmup_ratio 0.0 --logging_steps 20 \
    --allow_cls_grad \
    --max_n_train_example 1000 \
```

**Comment by frankaging** [comment]
@NitayGitHub Hey, thanks for raising the q.

A few comments of GLUE evals:
- generally, it is less stable than other benchmarks we tried.
- yes, all scores reported in the paper is on test set.
- the results are aggregated by strictly following: train on train set, do the in-training model selection on the validation set performance (i.e., overfitting to the dev set here); and use the best checkpoint (i.e., the checkpoint with best validation performance) to evaluate on test set.
- repeat the process with different random seeds. the seeds controls how dev/test sets are created via random splitting.

If you follow these steps closely; or if you ran our original code with a few random seeds, you should get approximately the same results. Be aware that GLUE is, unfortunately, just super small and unstable, so you might end up with slightly better or worse results on different datasets, especially smaller ones.

**Comment by frankaging** [comment]
<img width="611" height="201" alt="Image" src="https://github.com/user-attachments/assets/85622fd3-ce63-49f2-8a99-07c163a16c5c" />

Details from Appendix D.1.
